tests/core/test_centroids.py

```diff
@@ -0,0 +1,155 @@
+import numpy as np
+import pandas as pd
+import pytest
+from anndata import AnnData
+from numpy.random import default_rng
+from spatialdata._core.centroids import get_centroids
+from spatialdata.models import Labels2DModel, Labels3DModel, TableModel, get_axes_names
+from spatialdata.transformations import Identity, get_transformation, set_transformation
+
+from tests.core.operations.test_transform import _get_affine
+
+RNG = default_rng(42)
+
+affine = _get_affine()
+
+
+@pytest.mark.parametrize("coordinate_system", ["global", "aligned"])
+@pytest.mark.parametrize("is_3d", [False, True])
+def test_get_centroids_points(points, coordinate_system: str, is_3d: bool):
+    element = points["points_0"]
+
+    # by default, the coordinate system is global and the points are 2D; let's modify the points as instructed by the
+    # test arguments
+    if coordinate_system == "aligned":
+        set_transformation(element, transformation=affine, to_coordinate_system=coordinate_system)
+    if is_3d:
+        element["z"] = element["x"]
```

Comment:
I personally am not a big fan of tests with many conditionals because I find them difficult to read. I think `parameterize` is better used when multiple inputs can be passed through the same code (you can pass the expected results  as input arguments as well).

I think this would be easier to read if there were two sets of tests, one for global and one for aligned. you can then pass in 2D and 3D points with their expected values as inputs. Something like

```python
@pytest.mark.parametrize("points,expected", [(points_2d, expected_points_2d), (points_3d, expected_points_3d)])
def test_get_centroids_points_global(points, expected):
    # do stuff

@pytest.mark.parametrize("points,expected", [(points_2d, expected_points_2d), (points_3d, expected_points_3d)])
def test_get_centroids_points_aligned(points, expected):
    # do stuff
```

That being said, I think it's fine if you prefer this style. I don't want to block the PR - just wanted to give an alternative.

Reply:
Thanks for the comment, you are right the logic of `global` vs `aligned` is different and it would have been better to split the tests. I In the context of this PR I prefer to keep it like this because this PR is blocking other PRs and we need to make a release, but for the next one I will try to reduce the use of `parametrize` to the cases that don't create code bifurcation.